### ISSUE TITLE
bug:  #27206: [44] is undefined.

### ISSUE THREAD

**Issue by gaetbout** [starkware-libs/cairo#3119]
# Bug Report

**Cairo version:**

Tested at [RC.0](https://github.com/starkware-libs/cairo/releases/tag/v1.0.0-rc0)
When doing this code:
```rust
fn index_of_min<impl TPartialEq: PartialEq<T>, impl TPartialOrd: PartialOrd<T>>(
        mut self: Span<T>
    ) -> Option<usize> {
  let mut current_index = 0;
  let mut min_index = 0;
  let mut min = match self.pop_front() {
      Option::Some(item) => *item,
      Option::None(_) => {
          return Option::None(());
      },
  };
  loop {
      match self.pop_front() {
          Option::Some(item) => {
              if *item < min {
                  min = *item;
                  current_index = current_index + 1;
                  min_index = current_index; 
              }
          },
          Option::None(_) => {
              break Option::Some(min_index);
          },
      };
  }
```
I'm running into:
```
Error: Failed setting up runner.

Caused by:
   #27206: [44] is undefined.
```

However if I invert the two lines in the middle it works:
```rust
Option::Some(item) => {
    if *item < min {
        min = *item;
        min_index = current_index + 1; 
        current_index = current_index + 1;
    }
},
```

I realise my code makes no sense 😅

**Comment by orizi** [comment]
this has something to do with loops automatically decides vars to extract - apparently the usage finding here is still somewhat buggy.

**Comment by gilbens-starkware** [comment]
Fixed in #3437.
